packages/bee-q/src/components/panel/bq-panel.tsx

```diff
@@ -0,0 +1,147 @@
+import { h, Component, Prop, Element, Method, Watch, EventEmitter, Event } from '@stencil/core';
+
+import { FloatingUIPlacement } from '../../services/interfaces';
+import { FloatingUI } from '../../services/libraries';
+
+/**
+ * @part base - The `div` element which acts as a container for the panel content
+ */
+@Component({
+  tag: 'bq-panel',
+  styleUrl: './scss/bq-panel.scss',
+  shadow: true,
+})
+export class BqPanel {
+  // Own Properties
+  // ====================
+
+  private panel: HTMLElement;
+  private floatingUI: FloatingUI;
+  private trigger: HTMLElement;
+
+  // Reference to host HTML element
+  // ===================================
+
+  @Element() el!: HTMLBqPanelElement;
+
+  // State() variables
+  // Inlined decorator, alphabetical order
+  // =======================================
+
+  // Public Property API
+  // ========================
+
+  /** Distance (px) between the panel and the trigger element. */
+  @Prop({ reflect: true }) distance?: number = 0;
+
+  /** Position of the panel */
+  @Prop({ reflect: true }) placement?: FloatingUIPlacement = 'bottom';
+
+  /** If true, panel is visible.
+   * You can toggle this attribute to show/hide the panel.
+   */
+  @Prop({ reflect: true }) open?: boolean = false;
+
+  /** If true, the scrollbar is visible.
+   * You can toggle this attribute to show/hide the scrollbar.
+   */
+  @Prop({ reflect: true }) scrollbar?: boolean = false;
```

Comment:
We need to add all available options by FloatingUI, also **we can pass the trigger element through a property instead** of using a method for it.

```suggestion
  /** Represents the distance (gutter or margin) between the panel and the trigger element. */
  @Prop({ reflect: true }) distance?: number = 4;

  /** If true, the panel will be visible. */
  @Prop({ reflect: true, mutable: true }) open?: boolean = false;

  /** Position of the panel */
  @Prop({ reflect: true }) placement?: FloatingUIPlacement = 'bottom';

  /** The trigger element for the panel */
  @Prop() triggerElement?: HTMLElement;

  /** Whether the panel should have the same width as the trigger element */
  @Prop({ reflect: true }) sameWidth?: boolean = false;

  /**  Represents the skidding between the panel and the trigger element. */
  @Prop({ reflect: true }) skidding?: number = 0;

  /** Defines the strategy to position the panel */
  @Prop({ reflect: true }) strategy?: 'fixed' | 'absolute' = 'fixed';
```